app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java

```diff
@@ -716,9 +717,14 @@ public SearchTeiModel transform(TrackedEntitySearchItem searchItem, @Nullable Pr
         SearchTeiModel searchTei = new SearchTeiModel();
         if (dbTei != null && dbTei.aggregatedSyncState() != State.RELATIONSHIP) {
             searchTei.setTei(dbTei);
-            List<Enrollment> enrollmentsInProgram = d2.enrollmentModule().enrollments()
-                    .byTrackedEntityInstance().eq(dbTei.uid())
-                    .byProgram().eq(selectedProgram.uid())
+            EnrollmentCollectionRepository enrollmentRepository = d2.enrollmentModule().enrollments()
+                    .byTrackedEntityInstance().eq(dbTei.uid());
+
+            if (selectedProgram != null) {
+                enrollmentRepository.byProgram().eq(selectedProgram.uid());
```

Comment:
I think you have to do this:
`enrollmentRepository = enrollmentRepository.byProgram()...
`

Reply:
you are right